### ISSUE TITLE
Bot stops working after latest plugin update

### ISSUE THREAD

**Issue by hansw1204** [OpenWonderLabs/homebridge-switchbot#898]
### Describe Your Problem

Bot says "pressing" in HomeKit but doesn't move. There is no response in Homebridge log. I also updated the Switchbot firmware which may contribute to the issue. However, even with the new firmware v2.14.0 of the plugin works as expected.

### Relevant log output

```shell
After updating to v4.0.0 Switchbot stops responding. When I press the appropriate button in the Home app to activate the bot there is no response.
```

### Config for homebridge-switchbot

```shell
{
    "name": "SwitchBot",
    "credentials": {
        "token": "correct token",
        "secret": "correct secret",
        "notice": "Keep your Token & Secret a secret!"
    },
    "options": {
        "devices": [
            {
                "deviceId": "64E833E4BE64",
                "configDeviceName": "SwitchBot Hub 2",
                "configDeviceType": "Hub 2",
                "lock": {
                    "activate_latchbutton": false
                },
                "history": true,
                "firmware": "1.5-1.0",
                "refreshRate": 360
            },
            {
                "deviceId": "D2353334687D",
                "configDeviceName": "SwitchBot Bot",
                "configDeviceType": "Bot",
                "connectionType": "OpenAPI",
                "bot": {
                    "mode": "press",
                    "deviceType": "door",
                    "allowPush": true,
                    "doublePress": 1
                },
                "lock": {
                    "activate_latchbutton": false
                },
                "firmware": "6.6",
                "refreshRate": 360,
                "external": false
            }
        ],
        "refreshRate": 360
    },
    "_bridge": {
        "username": "0E:60:3D:E2:5A:D4",
        "port": 34884
    },
    "platform": "SwitchBot"
}
```

### Screenshots

nothing to show because there is no response in the logs after activation. This is what got looks like in 2.14.
[1/31/2024, 3:46:43 PM] [SwitchBot] Bot: SwitchBot Bot Set TargetPosition: 100
[1/31/2024, 3:46:55 PM] [SwitchBot] Bot: SwitchBot Bot Set TargetPosition: 0
[1/31/2024, 3:53:54 PM] [BathR Heater Timer] Setting switch to false
[1/31/2024, 3:55:30 PM] [SwitchBot] Bot: SwitchBot Bot Set TargetPosition: 100

### Device & Model

Hub 2 and Bot

### Node.js Version

v18.19.0

### NPM Version

v??

### Homebridge Version

v1.7.0

### Homebridge SwitchBot Plugin Version

v3.0.0.

### Homebridge Config UI X Plugin Version

v4.55.1

### Operating System

Docker

**Comment by donavanbecker** [comment]
turn debug logs on. If there is no logs I have nothing to work off of. Also I would recommend just using Hub 2 (Matter) with your bot as that is native connection.

**Comment by jeanchrijaz** [comment]
the same for me with 3.0.0... 
I downgraded from the previous version (v2.14.0) and works fine again

**Comment by ddong3000** [comment]
> 3.0.0에서도 마찬가지입니다. 이전 버전(v2.14.0)에서 다운그레이드했는데 잘 작동합니다.

me too!!!

**Comment by donavanbecker** [comment]
as stated before, I need logs.

**Comment by sonsongithub** [comment]
Me too, plugin's log is following,

On v3.0.0,

```
[01/02/2024, 11:05:09] [SwitchBot] Initializing SwitchBot platform...
```

This is all that is shown in the plugin log.
I also downgraded to 2.14.0 and works fine.

On v2.14.0, plugin's log shows following,

```
[01/02/2024, 11:10:46] [SwitchBot] Initializing SwitchBot platform...
[01/02/2024, 11:10:46] [SwitchBot] [DEBUG] Using debugMode Logging
[01/02/2024, 11:10:46] [SwitchBot] Finished initializing platform:
[01/02/2024, 11:10:46] [SwitchBot] [DEBUG] Using Default Refresh Rate (2 minutes).
[01/02/2024, 11:10:46] [SwitchBot] [DEBUG] Using Default Push Rate.
[01/02/2024, 11:10:46] [SwitchBot] Config OK
[01/02/2024, 11:10:46] [SwitchBot] Executed didFinishLaunching callback

/// continue....

```

**Comment by Z1mDMan** [comment]
[@switchbot_homebridge-switchbot.log.txt](https://github.com/OpenWonderLabs/homebridge-switchbot/files/14119715/%40switchbot_homebridge-switchbot.log.txt)

> the same for me with 3.0.0... I downgraded from the previous version (v2.14.0) and works fine again

same here. Hub Mini. logs attached. After upgrade I tried turning on a bulb in apple home and no logs generated.

**Comment by kellery82** [comment]
Same issue and when switching to Debug logging, nothing was shown and the switchbot was not working - i had to revert back to the previous version as well. I'm happy to provide other logs or do some testing if required.

**Comment by Exor** [comment]
Same issue here after upgrading to 3.0.0. I have a ceiling light, lightstrip, and bulbs that no longer respond. The only log entry is "[SwitchBot] Initializing SwitchBot platform..."
Platform: docker
Homebridge v1.7.0
Node.js  v18.19.0

**Comment by hansw1204** [comment]
I tried the native Matter connection with the previous Firmware and the performance was poor. The plugin provided crisp performance. I’ll try the Matter connection again with the new firmware. Maybe performance has improved.


From: Donavan Becker ***@***.***>
Date: Wednesday, January 31, 2024 at 16:46
To: OpenWonderLabs/homebridge-switchbot ***@***.***>
Cc: HaWi ***@***.***>, Author ***@***.***>
Subject: Re: [OpenWonderLabs/homebridge-switchbot] Bot stops working after latest plugin update (Issue #898)

turn debug logs on. If there is no logs I have nothing to work off of. Also I would recommend just using Hub 2 (Matter) with your bot as that is native connection.

**Comment by donavanbecker** [comment]
Is this only happening on Bots? Or are others having issues on other devices?

**Comment by Z1mDMan** [comment]
I have the issue with all switchbot devices in 3.0.0 and no issues if I roll back.

**Comment by Sarahlizzy** [comment]
Want to just note that it isn’t working for me either with 3.0. Reverted and it’s functioning  again. Literally nothing happens. No errors in the logs. It just doesn’t work.

**Comment by frankea** [comment]
Fresh install of homebridge and the plugin for my blind tilt and all I get even with debugging turned on is:

> [02/02/2024, 17:07:05] [SwitchBot] Initializing SwitchBot platform...
> [02/02/2024, 17:07:05] Loading 1 accessories...
> [02/02/2024, 18:40:03] [SwitchBot] Initializing SwitchBot platform...
> [02/02/2024, 18:40:03] Loading 1 accessories...
> [02/02/2024, 18:41:32] [SwitchBot] Initializing SwitchBot platform...
> [02/02/2024, 18:41:32] Loading 1 accessories...
> [02/02/2024, 18:42:43] [SwitchBot] Initializing SwitchBot platform...
> [02/02/2024, 18:42:43] Loading 1 accessories…

EDIT:  Installed v3.0.1-beta.1 and seeing the same thing:

> [02/02/2024, 20:43:04] [SwitchBot] Initializing SwitchBot platform...
> [02/02/2024, 20:43:04] Loading 1 accessories...

EDIT 2:

As with others, rolling back a version got everything working.

**Comment by donavanbecker** [comment]
Fixed with v3.1.0

**Comment by corasaniti** [comment]
@donavanbecker 
Fixed With 3.1 verssion, thanks

**Comment by hansw1204** [comment]
Confirmed, the issue is fixed in v3.1

From: Piero Corasaniti ***@***.***>
Date: Saturday, February 3, 2024 at 01:35
To: OpenWonderLabs/homebridge-switchbot ***@***.***>
Cc: HaWi ***@***.***>, Author ***@***.***>
Subject: Re: [OpenWonderLabs/homebridge-switchbot] Bot stops working after latest plugin update (Issue #898)

@donavanbecker<https://github.com/donavanbecker>
Fixed With 3.1 verssion, thanks
